src/components/CoursesGallery/CourseCard/CourseCard.jsx

```diff
@@ -1,77 +1,141 @@
+import { useEffect, useState } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import PropTypes from 'prop-types';
 import { Icon, Message, Title } from '../../common';
-import PropTypes from 'prop-types'
+import defaultImg from '../../../assets/images/default-avatar.png';
 import {
     Card,
     CardHeader,
     CardTitleWrapper,
     TitleBox,
     Avatar,
-    TimeInfo,
-    AboutTitle,
+    MobileSizeTimeInfo,
+    LargeSizeTimeInfo,
+    AdvertSubtitle,
     CourseDetails,
-    CourseDescr
+    CourseDescr,
+    CategoryContainer,
+    EducationContainer,
+    ContactsList,
 } from './CourseCard.styled';
-import { Link } from 'react-router-dom';
-
 
 
 const CourseCard = ({ advert }) => {
-    const isMobileScreenSize = window.matchMedia('(max-width: 768px)').matches;
-    const { id, title, attachment, created_at, description, experience, format, language, price, provider, time } = advert;
+    const { id, title, attachment, created_at, description, experience, format, language, price, provider, time, mobile_phone, telegram, email, category } = advert;
+    const pathname = useLocation().pathname;
+    const [isFullAdvertInfo, setIsFullAdvertInfo] = useState(false);
+
+
+    useEffect(() => {
+        if (pathname === `/courses/${id}`) {
+            setIsFullAdvertInfo(true);
+            return;
+        }
+        setIsFullAdvertInfo(false);
+    }, [pathname, id])
+    
+    const getAgeOfAdvert = () => {
```

Comment:
Since there are some calculations and dependencies here, I would suggest to use `useMemo` hook.

Reply:
has fixed